.jenkins/pytorch/win-test-helpers/build_pytorch.bat

```diff
@@ -122,7 +122,10 @@ if not "%USE_CUDA%"=="0" (
     copy %TMP_DIR_WIN%\bin\sccache.exe %TMP_DIR_WIN%\bin\nvcc.exe
   )
 
-  set CUDA_NVCC_EXECUTABLE=%TMP_DIR_WIN%\bin\nvcc
+  curl -kL https://github.com/peterjc123/randomtemp/releases/download/v0.2/randomtemp.exe --output %TMP_DIR_WIN%\bin\randomtemp.exe
+  set RANDOMTEMP_EXECUTABLE=%TMP_DIR_WIN%\bin\nvcc.exe
```

Comment:
Looks like the variable names `RANDOMTEMP_EXECUTABLE` and `CUDA_NVCC_EXECUTABLE ` are swapped?

Reply:
Previously, CMake uses `CUDA_NVCC_EXECUTABLE` for finding nvcc and then the calls are redirected to sccache. sccache looks for the actual `nvcc` in `PATH`, and then pass the arguments to it.
Currently, randomtemp is placed before sccache (%TMP_DIR_WIN%\bin\nvcc). So we are actually pretending sccache instead of nvcc itself.
I think it should be fine since `sccache` doesn't rely on `TEMP` or `TMP`. Furthermore, if we want to patch nvcc instead of sccache, then we may need some complicated changes. e.g. We will need to rename randomtemp to nvcc and change the PATH. We will also need to find out the actual PATH of nvcc. What do you think, @yf225 ?

Reply:
I don't mind the confusing naming as long as what you just wrote here is put in a comment here.

Reply:
Done.